### PR TITLE
Adding MiniBoss Alerts?

### PR DESCRIPTION
**Pull request by depth-dev**
i had a strokre

### CODE REVIEW

File: features/miniAlert.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            if(e.getName().includes("Lost Adventurer")) Client.showTitle("", "&dLost Adventurer", 0, 3, 0)
+            if(e.getName().includes("Angry Archeologist")) Client.showTitle("", "&cAngry Archeologist", 0, 3, 0)
+            if(e.getName().includes("Shadow Assassin")) Client.showTitle("", "&5Shadow Assassin", 0, 3, 0)
+            if(e.getName().includes("King Midas")) Client.showTitle("", "&6King Midas", 0, 3, 0)
+        }) 
+    }
+})
+
+//Thsi code is amazing and everoyne iwll like it
+//- misterdepth
```
Comment (Antonio32A) [inline features/miniAlert.js:25]: no newline at end of file

File: features/miniAlert.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//Thanks Vals
+function inDungeons() {
```
Comment (Antonio32A) [inline features/miniAlert.js:2]: should use an arrow function since you don't need the arguments

File: features/miniAlert.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+//Thanks Vals
+function inDungeons() {
+    if(!(Scoreboard.getTitle().trim() != "" && Scoreboard.getLines().length > 0)) {
```
Comment (Antonio32A) [inline features/miniAlert.js:3]: there should be a space between the if and the (),
yes this is a bit picky but the whole project uses it like that so you should too

File: features/miniAlert.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+//Thanks Vals
+function inDungeons() {
+    if(!(Scoreboard.getTitle().trim() != "" && Scoreboard.getLines().length > 0)) {
+        return;
+    }
+    for(let i = 0; i < Scoreboard.getLines().length; i++) {
```
Comment (Antonio32A) [inline features/miniAlert.js:6]: there should be a space between the for and the (),
yes this is a bit picky but the whole project uses it like that so you should too

File: features/miniAlert.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+//Thanks Vals
+function inDungeons() {
+    if(!(Scoreboard.getTitle().trim() != "" && Scoreboard.getLines().length > 0)) {
+        return;
+    }
+    for(let i = 0; i < Scoreboard.getLines().length; i++) {
+        let line = Scoreboard.getLines()[i]
+        if(line.getName().includes('§cThe Cata') && line.getName().includes('mbs §8('))
```
Comment (Antonio32A) [inline features/miniAlert.js:8]: there should be a space between the if and the (),
yes this is a bit picky but the whole project uses it like that so you should too

File: features/miniAlert.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    }
+    for(let i = 0; i < Scoreboard.getLines().length; i++) {
+        let line = Scoreboard.getLines()[i]
+        if(line.getName().includes('§cThe Cata') && line.getName().includes('mbs §8('))
+            return true
+    }
+    return false
+}
+register("renderWorld", () => {
+    if(settings.getSetting("Shit", "MoreShit") && inDungeons()) {
```
Comment (Antonio32A) [inline features/miniAlert.js:14]: the inDungeons should probably not be executed every frame
it should be executed every tick instead
I'm pretty sure that it's better for the performance

Comment (Antonio32A) [reply features/miniAlert.js:14]: also properly setup the settings

File: features/miniAlert.js (hunk 5 above)
Comment (Antonio32A) [inline features/miniAlert.js:8]: use double quotes since the whole project uses those so you should too

File: features/miniAlert.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+//Thanks Vals
```
Comment (Antonio32A) [inline features/miniAlert.js:1]: there should be a space after the // and before the text in the comment

File: features/miniAlert.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+//Thanks Vals
+function inDungeons() {
+    if(!(Scoreboard.getTitle().trim() != "" && Scoreboard.getLines().length > 0)) {
+        return;
+    }
+    for(let i = 0; i < Scoreboard.getLines().length; i++) {
+        let line = Scoreboard.getLines()[i]
```
Comment (Antonio32A) [inline features/miniAlert.js:7]: semicolon

File: features/miniAlert.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+function inDungeons() {
+    if(!(Scoreboard.getTitle().trim() != "" && Scoreboard.getLines().length > 0)) {
+        return;
+    }
+    for(let i = 0; i < Scoreboard.getLines().length; i++) {
+        let line = Scoreboard.getLines()[i]
+        if(line.getName().includes('§cThe Cata') && line.getName().includes('mbs §8('))
+            return true
+    }
+    return false
```
Comment (Antonio32A) [inline features/miniAlert.js:11]: there are semicolons missing

File: features/miniAlert.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+}
+register("renderWorld", () => {
+    if(settings.getSetting("Shit", "MoreShit") && inDungeons()) {
+        World.getAllEntities().forEach(e => {
+            if(e.getName().includes("Lost Adventurer")) Client.showTitle("", "&dLost Adventurer", 0, 3, 0)
```
Comment (Antonio32A) [inline features/miniAlert.js:16]: semicolon

File: features/miniAlert.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+            if(e.getName().includes("Angry Archeologist")) Client.showTitle("", "&cAngry Archeologist", 0, 3, 0)
```
Comment (Antonio32A) [inline features/miniAlert.js:17]: semicolon

File: features/miniAlert.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+            if(e.getName().includes("Shadow Assassin")) Client.showTitle("", "&5Shadow Assassin", 0, 3, 0)
```
Comment (Antonio32A) [inline features/miniAlert.js:18]: semicolon

File: features/miniAlert.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+            if(e.getName().includes("King Midas")) Client.showTitle("", "&6King Midas", 0, 3, 0)
```
Comment (Antonio32A) [inline features/miniAlert.js:19]: semicolon

File: features/miniAlert.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+        })
```
Comment (Antonio32A) [inline features/miniAlert.js:20]: semicolon

File: features/miniAlert.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+register("renderWorld", () => {
+    if(settings.getSetting("Shit", "MoreShit") && inDungeons()) {
+        World.getAllEntities().forEach(e => {
+            if(e.getName().includes("Lost Adventurer")) Client.showTitle("", "&dLost Adventurer", 0, 3, 0)
+            if(e.getName().includes("Angry Archeologist")) Client.showTitle("", "&cAngry Archeologist", 0, 3, 0)
+            if(e.getName().includes("Shadow Assassin")) Client.showTitle("", "&5Shadow Assassin", 0, 3, 0)
+            if(e.getName().includes("King Midas")) Client.showTitle("", "&6King Midas", 0, 3, 0)
+        }) 
+    }
+})
```
Comment (Antonio32A) [inline features/miniAlert.js:22]: semicolon

File: features/miniAlert.js [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+//Thsi code is amazing and everoyne iwll like it
```
Comment (Antonio32A) [inline features/miniAlert.js:24]: unnecessary comment

File: features/miniAlert.js (hunk 1 above)
Comment (Antonio32A) [inline features/miniAlert.js:25]: unnecessary comment, credit yourself in the settings file

File: features/miniAlert.js (hunk 6 above)
Comment (Antonio32A) [reply features/miniAlert.js:14]: oh also same applies to all the entity checking.

File: features/miniAlert.js (hunk 7 above)
Comment (Antonio32A) [reply features/miniAlert.js:1]: you should add this into a class, add proper settings and add the file to the loader so it gets loaded

**Review by Antonio32A — CHANGES_REQUESTED** [review]
good luck

File: features/miniAlert.js (hunk 1 above)
Comment (Debuggingss) [reply features/miniAlert.js:25]: Didn't Ask.

File: features/miniAlert.js (hunk 2 above)
Comment (Debuggingss) [reply features/miniAlert.js:2]: Didn't Ask.

File: features/miniAlert.js (hunk 3 above)
Comment (Debuggingss) [reply features/miniAlert.js:3]: Didn't Ask.

File: features/miniAlert.js (hunk 4 above)
Comment (Debuggingss) [reply features/miniAlert.js:6]: Didn't Ask.

File: features/miniAlert.js (hunk 5 above)
Comment (Debuggingss) [reply features/miniAlert.js:8]: Didn't Ask.

File: features/miniAlert.js (hunk 6 above)
Comment (Debuggingss) [reply features/miniAlert.js:14]: Didn't Ask.

File: features/miniAlert.js (hunk 5 above)
Comment (Debuggingss) [reply features/miniAlert.js:8]: Didn't Ask.

File: features/miniAlert.js (hunk 7 above)
Comment (Debuggingss) [reply features/miniAlert.js:1]: Didn't Ask.

File: features/miniAlert.js (hunk 8 above)
Comment (Debuggingss) [reply features/miniAlert.js:7]: Didn't Ask.

File: features/miniAlert.js (hunk 10 above)
Comment (Debuggingss) [reply features/miniAlert.js:16]: Didn't Ask.

File: features/miniAlert.js (hunk 9 above)
Comment (Debuggingss) [reply features/miniAlert.js:11]: Didn't Ask.

File: features/miniAlert.js (hunk 1 above)
Comment (Debuggingss) [reply features/miniAlert.js:25]: Didn't Ask.

File: features/miniAlert.js (hunk 16 above)
Comment (Debuggingss) [reply features/miniAlert.js:24]: Didn't Ask.

File: features/miniAlert.js (hunk 15 above)
Comment (Debuggingss) [reply features/miniAlert.js:22]: Didn't Ask.

File: features/miniAlert.js (hunk 14 above)
Comment (Debuggingss) [reply features/miniAlert.js:20]: Didn't Ask.

File: features/miniAlert.js (hunk 13 above)
Comment (Debuggingss) [reply features/miniAlert.js:19]: Didn't Ask.

File: features/miniAlert.js (hunk 12 above)
Comment (Debuggingss) [reply features/miniAlert.js:18]: Didn't Ask.

File: features/miniAlert.js (hunk 11 above)
Comment (Debuggingss) [reply features/miniAlert.js:17]: Didn't Ask.

**Comment by Antonio32A** [comment]
moved to https://github.com/Antonio32A/DungeonUtilities/pull/4 ig

**Pull request closed without merging**